### ISSUE TITLE
zoltan2: FindTPLParMETIS.cmake does not require metis.h

### ISSUE THREAD

**Issue by tjfulle** [trilinos/Trilinos#2199]
@trilinos/zoltan2 : `FindTPLParMETIS.cmake` fails to find parmetis if metis is installed to a different installation prefix than parmetis.  parmetis is actually found, but fails the configure-time check of its version because the header `metis.h` cannot be found by the compiler.  Replacing

```
TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( ParMETIS
  REQUIRED_HEADERS parmetis.h
  REQUIRED_LIBS_NAMES "parmetis;metis"
  )
```

with

```
TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( ParMETIS
  REQUIRED_HEADERS "parmetis.h;metis.h"
  REQUIRED_LIBS_NAMES "parmetis;metis"
  )
```

fixes the problem.

This is important for TPLs installed by spack, since spack installs software packages to their own (independent) installation prefixes.

**Comment by tjfulle** [comment]
@trilinos/zoltan2, the fix is so simple I could submit a PR in just a few minutes, I just wasn't sure if there was a specific reason for not requiring the metis header.

**Comment by kddevin** [comment]
This change is fine with me as long as it is OK with @trilinos/framework .
Thanks.

**Comment by bmpersc** [comment]
@tjfulle and @kddevin I'm trying to recall if this was done on purpose due to issues with mismatched parmetis and metis libraries being found. I know that those two libraries have caused us a lot of heartache in the past with versioning and incompatibilities. I'm not sure how relevant that is today. We are trying to increase our TPL usage for testing and both parmetis and metis will be available to us. Looking at the log suggests that the only header for the parmetis library that we've ever searched for so this could easily be an oversight.

Given the change to most of our testing environments to use modules instead of having to setup an environment for all tests I think that the change has very little chance to break our testing. So I'm okay with the change.

**Comment by tjfulle** [comment]
Fixed in #2302
